tla/consensus/abs.tla

```diff
@@ -0,0 +1,58 @@
+---- MODULE abs ----
+\* Abstract specification for a distributed consensus algorithm.
+\* Assumes that any node can inspect the state of all other nodes. 
+
+EXTENDS Sequences, SequencesExt, Naturals, FiniteSets
+
+CONSTANT Servers, Terms, MaxLogLength, StartTerm
+
+\* Commit logs from each node
+\* Each log is append-only and the logs will never diverge.
+VARIABLE CLogs
+
+
+InitialLogs == {
+    <<>>,
+    <<StartTerm, StartTerm>>,
+    <<StartTerm, StartTerm, StartTerm, StartTerm>>}
+    
+Init ==
+    CLogs \in [Servers -> InitialLogs]
+
+\* A node i can copy a ledger suffix from another node j.
+Copy(i) ==
+    \E j \in Servers : 
+        /\ Len(CLogs[j]) > Len(CLogs[i])
+        /\ \E l \in 1..(Len(CLogs[j]) - Len(CLogs[i])) : 
+                CLogs' = [CLogs EXCEPT ![i] = CLogs[i] \o SubSeq(CLogs[j], Len(CLogs[i]) + 1, Len(CLogs[i]) + l)]
+
+\* The node with the longest log can extend its log.
+Extend(i) ==
+    /\ \A j \in Servers : Len(CLogs[j]) \leq Len(CLogs[i])
+    /\ \E l \in 0..(MaxLogLength - Len(CLogs[i])) : 
```

Comment:
Given that it's state-space is small, I'd suggest to conceptually move `MaxLogLength` into a state constraint to improve the clarity of this spec.

Reply:
Is this (https://github.com/microsoft/CCF/pull/6438/commits/8931e18313570a194edaf7a648d434f8556f283b) what you mean or did you mean something different?